File: src/main/java/dev/algos/snatch/interview_problems/array/MaxScore.py
from typing import List


class Solution:
    def maxScore(self, arr: List[int], k: int) -> int:
        sum = 0
        size = len(arr) - k
        for i in range(0, size):
            sum += arr[i]
        min = sum
        total = sum
        for i in range(size, len(arr)):
            total += arr[i]
            sum -= arr[i - size]
            sum += arr[i]
            if sum < min:
                min = sum
        return total - min


Solution().maxScore([1, 2, 3, 4, 5, 6, 1], 3)

File: src/main/java/dev/algos/snatch/interview_problems/array/max_score.rs
pub fn max_score(arr: Vec<i32>, k: i32) -> i32 {
    let mut sum = 0;
    for i in 0..arr.len() {
        sum += arr[i];
    }
    let mut total = sum;
    let mut min = sum;
    for i in arr.len() - k..arr.len() {
        sum += arr[i];
        sum -= arr[i - arr.len() - k];
        if sum < min {
            min = sum;
        }
    }
    return total - min;
}
